pkg/controller/bookkeepercluster/bookkeepercluster_controller.go

```diff
@@ -234,34 +235,52 @@ func (r *ReconcileBookkeeperCluster) syncBookieSize(bk *bookkeeperv1alpha1.Bookk
 func (r *ReconcileBookkeeperCluster) reconcileFinalizers(bk *bookkeeperv1alpha1.BookkeeperCluster) (err error) {
 	if bk.DeletionTimestamp.IsZero() {
 		if !util.ContainsString(bk.ObjectMeta.Finalizers, util.ZkFinalizer) {
-			bk.ObjectMeta.Finalizers = append(bk.ObjectMeta.Finalizers, util.ZkFinalizer)
+			// appending name of pravega cluster to the finalizers
+			finalizer := util.ZkFinalizer
+			configMap := &corev1.ConfigMap{}
+			if strings.TrimSpace(bk.Spec.EnvVars) != "" {
+				err = r.client.Get(context.TODO(), types.NamespacedName{Name: strings.TrimSpace(bk.Spec.EnvVars), Namespace: bk.Namespace}, configMap)
+				if err == nil {
+					clusterName, ok := configMap.Data["PRAVEGA_CLUSTER_NAME"]
+					if ok {
+						finalizer = finalizer + "_" + clusterName
+					}
+				}
+			}
+			bk.ObjectMeta.Finalizers = append(bk.ObjectMeta.Finalizers, finalizer)
 			if err = r.client.Update(context.TODO(), bk); err != nil {
 				return fmt.Errorf("failed to add the finalizer (%s): %v", bk.Name, err)
 			}
 		}
 	} else {
 		if util.ContainsString(bk.ObjectMeta.Finalizers, util.ZkFinalizer) {
-			bk.ObjectMeta.Finalizers = util.RemoveString(bk.ObjectMeta.Finalizers, util.ZkFinalizer)
+			finalizer := util.GetString(bk.ObjectMeta.Finalizers, util.ZkFinalizer)
+			pravegaClusterName := strings.Replace(finalizer, util.ZkFinalizer, "", 1)
```

Comment:
Move the lines for getting cluster name from slice inside another method and invoke that here.

Reply:
done